include/rocksdb/options.h

```diff
@@ -888,6 +888,16 @@ struct DBOptions {
   // Immutable.
   bool allow_ingest_behind = false;
 
+  // Needed to support differential snapshots.
+  // If set to true then DB will only process deletes with sequence number
+  // less than what was set by SetPreserveDeletesSequenceNumber(uint64_t ts).
+  // Clients are responsible to periodically call this method to advance
+  // the cutoff time. If this method is never called and preserve_deletes
+  // is set to true NO deletes will ever be processed.
+  // DEFAULT: false
+  // Immutable (TODO: make it dynamically changeable)
+  bool preserve_deletes = false;
```

Comment:
I think it would open up a possibility to undesirable scenarios like this:

 - DB.Open() is called.
 - SetPreserveDeletesSequenceNumber(1) is called
 - some data is added, some data is removed, but all deletes are preserved
 - DB.Close()
 - DB.Open()

Now do we expect people to call SetPreserveDeletesSequenceNumber(1) again after DB.Open() returns? (also since we're going to store the actual cutoff value in the RocksDB itself as a special key, we'll need to read it back first) Does it leave a gap when some deletes could be dropped by an eager automatically scheduled compaction?

Having an option seems safer way to me and easier to reason about the DB state, though I totally understand that I've just added new DB option :)

Reply:
Right now the way it works is that the `preserve_deletes_seqnum_`  in the DB is defaulted to 0; which means if `preserve_deletes` DB options was set to `true`, all deletes are preserved until user calls `SetPreserveDeletesSequenceNumber(some_seqnum)`. That seems safer to me.

Reply:
yes, the behavior you described sounds more convenient to me too, compared to user having to put a line immediately after `DB::Open` to start preserving deletes. Let's leave this new option as you have it :).